api/lib/gateways/UHW/UHWSearchGateway.js

```diff
@@ -0,0 +1,80 @@
+const path = require('path');
+const { loadSQL } = require('../../Utils');
+const { Systems } = require('../../Constants');
+const { searchCustomersBaseSQL } = loadSQL(path.join(__dirname, 'sql'));
+
+module.exports = options => {
+  const db = options.db;
+  const buildSearchRecord = options.buildSearchRecord;
+
+  const search = async queryParams => {
+    let whereClause = [];
+    let params = [];
+
+    if (queryParams.firstName && queryParams.firstName !== '') {
+      params.push({
+        id: 'forename',
+        type: 'NVarChar',
+        value: `%${queryParams.firstName.toLowerCase()}%`
+      });
+      whereClause.push(
+        'Forenames collate SQL_Latin1_General_CP1_CI_AS LIKE @forename'
+      );
+    }
+
+    if (queryParams.lastName && queryParams.lastName !== '') {
+      params.push({
+        id: 'surname',
+        type: 'NVarChar',
+        value: `%${queryParams.lastName.toLowerCase()}%`
+      });
+      whereClause.push(
+        'Surname collate SQL_Latin1_General_CP1_CI_AS LIKE @surname'
+      );
+    }
+
+    whereClause = whereClause.map(clause => `(${clause})`);
+
+    let query = `${searchCustomersBaseSQL} WHERE (${whereClause.join(
```

Comment:
Let's try use const in places like this to get into good habits! (I know you didn't write this :-))